pkg/agent/reconcilers.go

```diff
@@ -124,65 +123,51 @@ func (a *Agent) handleAppliedManifestWork(obj runtime.Object, isBeingDeleted boo
 			return false, nil
 		}
 		appliedManifestWorkInfo := appliedManifestWorkInfoIntf.(util.AppliedManifestInfo)
-		a.trackedObjects.RemoveTrackedObjectsUID(appliedManifestWorkInfo.ObjectUIDs)
 		a.stopInformers(appliedManifestWorkInfo)
 		a.trackedAppliedManifests.Delete(mObj.GetName())
 	}
 
 	return false, nil
 }
 
-func (a *Agent) updateWorkStatus(obj runtime.Object, isBeingDeleted bool) error {
+func (a *Agent) updateWorkStatus(obj runtime.Object) error {
 	mObj := obj.(metav1.Object)
 	namespace := a.clusterName
 
-	workStatus := &v1alpha1.WorkStatus{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      util.BuildWorkstatusName(obj),
-			Namespace: namespace,
-		},
-	}
-
 	ctx, cancel := context.WithTimeout(context.Background(), 2*time.Minute)
 	defer cancel()
 
-	// delete WorkStatus if exists, when the workload object is deleted
-	if isBeingDeleted {
-		err := a.hubClient.Get(ctx, client.ObjectKeyFromObject(workStatus), workStatus, &client.GetOptions{})
-		if err != nil && !apierrors.IsNotFound(err) {
-			return err
-		}
-		err = a.hubClient.Delete(ctx, workStatus, &client.DeleteOptions{})
```

Comment:
If we remove this, how do we delete a `WorkStatus` when one manifest was removed from the list of manifests in a `ManifestWork`? With multi-wrapped objects we should have the ability to explicitly delete a `WorkStatus`, the GC will remove it only when the whole `ManifestWork` will be removed(maybe never).

Reply:
good catch @andreyod  - will bring back the delete code.

Reply:
I have also added a test to verify that deletion of an object within a manifest causes the deletion of the corresponding work status